src/Entity/IdeasWorkshop/Idea.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+namespace AppBundle\Entity\IdeasWorkshop;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Committee;
+use AppBundle\Entity\EntityIdentityTrait;
+use AppBundle\Entity\EntityNameSlugTrait;
+use AppBundle\Entity\EntityTimestampableTrait;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity
+ *
+ * @ORM\Table(
+ *     name="ideas_workshop_idea",
+ *     uniqueConstraints={
+ *         @ORM\UniqueConstraint(name="idea_slug_unique", columns="slug")
+ *     }
+ * )
+ *
+ * @UniqueEntity("slug")
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class Idea
+{
+    use EntityIdentityTrait;
+    use EntityTimestampableTrait;
+    use EntityNameSlugTrait;
+
+    private const PUBLISHED_INTERVAL = 'P3W';
+
+    /**
+     * @ORM\ManyToOne(targetEntity="Theme")
+     */
+    private $theme;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="Category")
+     */
+    private $category;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="Need")
+     * @ORM\JoinTable(name="ideas_workshop_ideas_needs")
+     */
+    private $needs;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Adherent")
+     */
+    private $adherent;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime", nullable=true)
+     */
+    private $publishedAt;
+
+    /**
+     * @var Committee
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Committee")
+     */
+    private $committee;
+
+    /**
+     * @Assert\Choice(
+     *     callback={"AppBundle\Entity\IdeasWorkshop\IdeaStatusEnum", "toArray"},
+     *     strict=true,
+     * )
+     *
+     * @ORM\Column(length=11, options={"default": IdeaStatusEnum::IN_PROGRESS})
+     */
+    private $status;
+
+    /**
+     * @ORM\OneToMany(targetEntity="Answer", mappedBy="idea")
+     */
+    private $answers;
+
+    public function __construct(
+        UuidInterface $uuid,
+        string $name,
+        Adherent $adherent,
+        Category $category,
+        Theme $theme,
+        ?Committee $committee,
+        ?\DateTime $publishedAt = null,
```

Comment:
with `= null` the `?` become useless, you can write `\DateTime $date = null`

Reply:
:heavy_check_mark: